### ISSUE TITLE
Support CONNECT method for HTTP proxy

### ISSUE THREAD

**Issue by Fangliding** [telegramdesktop/tdesktop#28808]
### Is your feature request related to a problem?

I think this is indeed a problem

### Describe the solution you'd like

When using HTTP proxy in tdesktop, it only attempts to send data using the plain HTTP method (POST), which generates a huge number of connections (approximately several per second) that are not friendly to many proxies
But when using socks, it will only open a few TCP relays normally and transmit within them
HTTP proxy also supports direct relay of TCP (using CONNECT method), which can improve the efficiency of the proxy

### Describe alternatives you've considered

The solution is very obvious, there is no alternatives :)

### Additional context

I have read the mtp protocol and I think the HTTP transport is designed to adapt to the browser's request/response HTTP model, which is unnecessary on tdesktop

**Comment by ilya-fedin** [comment]
proxy support is done by qt, not tdesktop

**Comment by Fangliding** [comment]
> http/socks proxy support is done by qt, not tdesktop

I think the problem is here
https://github.com/telegramdesktop/tdesktop/blob/4505a2bf2dbb73185b4bb8b18b2aab721765bb7e/Telegram/SourceFiles/mtproto/session.cpp#L244
It disabled TCP when using HTTP proxy

**Comment by ilya-fedin** [comment]
Can you rephrase the title as "Use MTPROTO-over-TCP for HTTP proxy"?
